src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/FaultDetectionMetricsProcessor.java

```diff
@@ -0,0 +1,142 @@
+            FaultDetectionMetricsSnapshot faultDetectionMetricsSnapshot =
+                    new FaultDetectionMetricsSnapshot(conn, currWindowStartTime);
+            Map.Entry<Long, FaultDetectionMetricsSnapshot> entry = faultDetectionMetricsMap.lastEntry();
+            if (entry != null) {
+                faultDetectionMetricsSnapshot.rolloverInFlightRequests(entry.getValue());
+            }
+            faultDetectionMetricsMap.put(currWindowStartTime, faultDetectionMetricsSnapshot);
+            return new FaultDetectionMetricsProcessor(faultDetectionMetricsSnapshot);
+        } else {
+            return new FaultDetectionMetricsProcessor(faultDetectionMetricsMap.get(currWindowStartTime));
+        }
+    }
+
+    @Override
+    public void initializeProcessing(long startTime, long endTime) {
+        this.startTime = startTime;
+        this.endTime = endTime;
+        this.handle = faultDetectionMetricsSnapshot.startBatchPut();
+    }
+
+    @Override
+    public void finalizeProcessing() {
+        if (handle.size() > 0) {
+            handle.execute();
+        }
+        LOG.debug("Final Fault Detection request metrics {}", faultDetectionMetricsSnapshot.fetchAll());
+    }
+
+    @Override
+    public void processEvent(Event event) {
+        String[] keyItems = event.key.split(File.separatorChar == '\\' ? "\\\\" : File.separator);
+        if (keyItems[0].equals(PerformanceAnalyzerMetrics.sFaultDetection)) {
+            if (keyItems[3].equals(PerformanceAnalyzerMetrics.START_FILE_NAME)) {
+                emitStartMetric(event, keyItems);
+            } else if (keyItems[3].equals(PerformanceAnalyzerMetrics.FINISH_FILE_NAME)) {
+                emitFinishMetric(event, keyItems);
+            }
+        }
+    }
+
+    @Override
+    public boolean shouldProcessEvent(Event event) {
+        return event.key.contains(PerformanceAnalyzerMetrics.sFaultDetection);
+    }
+
+    @Override
+    public void commitBatchIfRequired() {
+        if (handle.size() > BATCH_LIMIT) {
+            handle.execute();
+            handle = faultDetectionMetricsSnapshot.startBatchPut();
+        }
+    }
+
+    // A keyItem is of the form : [fault_detection, follower_check, 76532, start]
+    //
+    // Example value part of the entry is:
+    // current_time:1566413979979
+    // StartTime:1566413987986
+    // SourceNodeID:g52i9a93a762cd59dda8d3379b09a752a
+    // TargetNodeID:b2a5a93a762cd59dda8d3379b09a752a
+    // $
+    private void emitStartMetric(Event entry, String[] keyItems) {
+        Map<String, String> keyValueMap = ReaderMetricsProcessor.extractEntryData(entry.value);
+
+        String sourceNodeId = keyValueMap.get(AllMetrics.FaultDetectionDimension.SOURCE_NODE_ID.toString());
+        String targetNodeId = keyValueMap.get(AllMetrics.FaultDetectionDimension.TARGET_NODE_ID.toString());
+        String startTimeVal = keyValueMap.get(AllMetrics.CommonMetric.START_TIME.toString());
```

Comment:
Import `AllMetrics.FaultDetectionDimension` instead of `AllMetrics` ?

Reply:
done